torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -527,86 +527,83 @@ def prepare_for_partitioner(mod, num_primals, num_fw_outputs):
             and isinstance(node.args[1], str)
         ):
-            identifier = (
-                node.args[1].replace("___forward", "").replace("___backward", "")
-            )
-
-            # NB: This is done in a separate if else condition because we early
-            # return if the partitioning_done is True.
             if node.args[1].startswith("___forward"):
                 fw_hop_nodes.append(node)
             elif node.args[1].startswith("___backward"):
-                found_backward_node = True
                 bw_hop_nodes.append(node)
 
-            # If partitioning already done for this identifier, skip. This saves
-            # redundant joint graph passes for same subgraphs.
-            if new_hop_graphs[identifier].partitioning_done:
-                continue
-
-            hop_gm = getattr(joint_gm, node.args[0].target)
-            assert isinstance(hop_gm, torch.fx.GraphModule)
-
-            if node.args[1].startswith("___forward"):
-                # Collect some information from the forward hop graph
-                new_hop_graphs[identifier].old_num_fw_inputs = num_inputs(hop_gm)
-                new_hop_graphs[identifier].old_num_fw_outputs = num_outputs(hop_gm)
-            elif node.args[1].startswith("___backward"):
-                num_fw_inputs = new_hop_graphs[identifier].old_num_fw_inputs
-                assert num_fw_inputs is not None
-                num_fw_outputs = new_hop_graphs[identifier].old_num_fw_outputs
-                assert num_fw_outputs is not None
-
-                # Step 1) - Get the `joint_hop_gm`. As mentioned earlier, the
-                # backward graph is the joint graph.
-                joint_hop_gm = hop_gm
-
-                # Prepare the graph for the partitioner
-                joint_hop_gm = prepare_for_partitioner(
-                    joint_hop_gm, num_fw_inputs, num_fw_outputs
-                )
+    if not bw_hop_nodes:
+        return joint_gm
 
-                # TODO: invoke_subgraph should track which of its inputs static indices
-                # so it can propagate them to the partitioner (and use in cudagraphs)
-                static_lifetime_input_indices: list[int] = []
-                # Step 2) and 3) - Run joint graph passes and partitioner
-                new_fw_hop_gm, new_bw_hop_gm = aot_config.partition_fn(
-                    joint_hop_gm,
-                    [],
-                    num_fwd_outputs=num_fw_outputs,
-                    static_lifetime_input_indices=static_lifetime_input_indices,
-                )
+    assert len(fw_hop_nodes) == len(bw_hop_nodes)
+
+    # Create a bw to hop node mapping. This helps us in identifying the bw and
+    # fw subgraph pairs without relying on the identifier. This is important
+    # because we can have different subgraphs for bwd for same subgraph in the
+    # fwd because of differing strides in the backward.
+    bw_to_fw_hop_node = dict(zip(list(reversed(bw_hop_nodes)), fw_hop_nodes))
```

Comment:
```suggestion
    bw_to_fw_hop_node = dict(zip(reversed(bw_hop_nodes), fw_hop_nodes))
```
List call is pointless, zip takes iterators including reversed ones